### ISSUE TITLE
XCM emulator docs and examples

### ISSUE THREAD

**Issue by xlc** [paritytech/cumulus#2986]
Right now it have pretty much no docs and many small issues, which can be discovered if someone is trying to use it to create a simple example. I have fixed all the issues I discovered myself with https://github.com/paritytech/cumulus/pull/2985

**Comment by NachoPal** [comment]
@xlc You are right, good documentation is needed. Since it was still under some heavy development/improvement, we didn't find yet time for it.

A developers guide will be created.
